PowerEditor/installer/nppSetup.nsi

```diff
@@ -270,64 +270,33 @@ Var nppSubStr
 
 ${MementoSection} "Context Menu Entry" explorerContextMenu
 
-	${If} $WinVer == "11"
-	
-		; Clean up the hack of v8.5 installer
-		ReadRegStr $muiVerbStr HKLM "SOFTWARE\Classes\*\shell\pintohome" MUIVerb
-		${StrStr} $nppSubStr $muiVerbStr "Notepad++"
-		; Make sure there's an entry, and the entry belong to Notepad++ before deleting it
-		${If} $muiVerbStr != ""
-			${AndIf} $nppSubStr != ""  ; it contains "Notepad++"
-				DeleteRegKey HKLM "SOFTWARE\Classes\*\shell\pintohome"
-		${EndIf}
-		
-		; Install the new Windows 11 "Edit with Notepad++" menu entry
-		!ifdef ARCHARM64
-			File /oname=$INSTDIR\NppModernShell.msix "..\binarm64\NppModernShell.msix"
-			File /oname=$INSTDIR\NppModernShell.dll "..\binarm64\NppModernShell.dll"
-		!else ; !ifdef ARCH64
-			File /oname=$INSTDIR\NppModernShell.msix "..\bin64\NppModernShell.msix"
-			File /oname=$INSTDIR\NppModernShell.dll "..\bin64\NppModernShell.dll"
-		!endif
-		Exec 'rundll32.exe "$INSTDIR\NppModernShell.dll",RegisterSparsePackage'
-
-		; Make sure old NppShell dll's are unregistered and removed
-		Exec 'regsvr32 /u /s "$INSTDIR\NppShell_01.dll"'
-		Exec 'regsvr32 /u /s "$INSTDIR\NppShell_02.dll"'
-		Exec 'regsvr32 /u /s "$INSTDIR\NppShell_03.dll"'
-		Exec 'regsvr32 /u /s "$INSTDIR\NppShell_04.dll"'
-		Exec 'regsvr32 /u /s "$INSTDIR\NppShell_05.dll"'
-		Exec 'regsvr32 /u /s "$INSTDIR\NppShell_06.dll"'
-		Delete "$INSTDIR\NppShell_01.dll"
-		Delete "$INSTDIR\NppShell_02.dll"
-		Delete "$INSTDIR\NppShell_03.dll"
-		Delete "$INSTDIR\NppShell_04.dll"
-		Delete "$INSTDIR\NppShell_05.dll"
-		Delete "$INSTDIR\NppShell_06.dll"
-		
-	${Else} ; the old "Edit with Notepad++" menu entry still works under Windows 10 and previous OS
-	
-		SetOverwrite try
-		SetOutPath "$INSTDIR\"
-		
-		; There is no need to keep x86 NppShell_06.dll in 64 bit installer
-		; But in 32bit installer both the Dlls are required
-		; As user can install 32bit npp version on x64 bit machine, that time x64 bit NppShell is required.
-		
-		!ifdef ARCH64
-			File /oname=$INSTDIR\NppShell_06.dll "..\bin\NppShell64_06.dll"
-		!else ifdef ARCHARM64
-			File /oname=$INSTDIR\NppShell_06.dll "..\binarm64\NppShell64.dll"
-		!else
-			${If} ${RunningX64}
-				File /oname=$INSTDIR\NppShell_06.dll "..\bin\NppShell64_06.dll"
-			${Else}
-				File "..\bin\NppShell_06.dll"
-			${EndIf}
-		!endif
-		Exec 'regsvr32 /s "$INSTDIR\NppShell_06.dll"'
-
-	${EndIf}
+	SetOverwrite try
+	SetOutPath "$INSTDIR\"
+
+	!ifdef ARCH64
+		File /oname=$INSTDIR\NppShell.msix "..\binarm64\NppShell.msix"
+		File /oname=$INSTDIR\NppShell.x64.dll "..\binarm64\NppShell.dll"
+	!else ifdef ARCHARM64
+		File /oname=$INSTDIR\NppShell.msix "..\binarm64\NppShell.msix"
+		File /oname=$INSTDIR\NppShell.arm64.dll "..\binarm64\NppShell.dll"
+	!else
+		File /oname=$INSTDIR\NppShell.x86.dll "..\binarm64\NppShell.dll"
+	!endif
+	Exec 'regsvr32 /s "$INSTDIR\NppShell.dll"'

```

Comment:
Does it work also under Windows11?

Reply:
I just tested by compiling the project into the x64 binary, and put both NppShell.x64.dll (renamed to NppShell.dll) & NppShell.msix into `C:\Program Files\Notepad++\`, then run `regsvr32 /s .\NppShell.dll` under the powershell - nothing happens. I then restarted explorer by typing `Get-process explorer | stop-process` - there's always no "Edit with Notepad++" on context menu of Windows 11, nor on the old context menu (Show more options).

Am I missing somethings?

Reply:
Can you try without the `/s` and see if it shows a dialog saying that it finished running.
If it doesn't, then the code is crashing somewhere.

Reply:
I just tested on a clean machine, installed npp 8.5, copied the dll and msix, installed my test cert, and tested with regsvr32 on it.
So it should be working, but there might be something that I have missed, that is causing it to fail (since I always test on a clean sandbox machine)

Reply:
@GurliGebis 
Nm, my bad - I've just forgotten to sign the binaries - after signing it works like a charm!
Sorry for the false alert.

Reply:
So both old (Show more options) and new context menu of Windows 11 have the "Edit with Notepad++" entry - is it intentional (or rather normal behaviour) ?

Reply:
> It starts by unregistering both the new and old kind, and then registering the sparse package for Win11, and the old context menu for anything below Win11 

Sorry, I didn't read it - I guess the answer is it :)

Reply:
Yep, the new one also shows up in the old menu on Windows 11 🙂

Reply:
No worries 🙂 I like how this cleans up the install script.

Reply:
So `NppShell.msix` is only for 64 bits binaries?